src/wkb.js

```diff
@@ -0,0 +1,206 @@
+/**
+ * WKB (Well-Known Binary) decoder for geometry objects.
+ *
+ * @import {DataReader} from '../src/types.d.ts'
+ * @import {Geometry} from '../src/types.js'
+ * @param {DataReader} reader
+ * @returns {Geometry} geometry object
+ */
+export function decodeWKB(reader) {
+  const { view } = reader
+  const isLittleEndian = view.getUint8(reader.offset++) === 1
+
+  // Read geometry type
+  const rawGeometryType = view.getUint32(reader.offset, isLittleEndian)
+  reader.offset += 4
+  const { type: geometryType, hasZ, hasM, hasSRID } = getFlags(rawGeometryType)
+
+  if (hasSRID) {
+    reader.offset += 4 // SRID is int32
+  }
+
+  if (geometryType === 1) { // Point
+    return { type: 'Point', coordinates: readPoint(reader, isLittleEndian, hasZ, hasM) }
+  } else if (geometryType === 2) { // LineString
+    const numPoints = view.getUint32(reader.offset, isLittleEndian)
+    reader.offset += 4
+    const coords = new Array(numPoints)
+    for (let i = 0; i < numPoints; i++) {
+      coords[i] = readPoint(reader, isLittleEndian, hasZ, hasM)
+    }
+    return { type: 'LineString', coordinates: coords }
+  } else if (geometryType === 3) { // Polygon
+    const numRings = view.getUint32(reader.offset, isLittleEndian)
+    reader.offset += 4
+    const coords = new Array(numRings)
+    for (let r = 0; r < numRings; r++) {
+      const numPoints = view.getUint32(reader.offset, isLittleEndian)
```

Comment:
the standard expects the number of points to be at least 4, and the first and last points must be the same. Should we enforce it? Or is it ok to produce invalid geojson if the data is not well formed?

Reply:
If the input file is invalid, then it is [Undefined Behavior](https://en.wikipedia.org/wiki/Undefined_behavior).
My philosophy is that we should not validate or enforce correctness unnecessarily.